samples/client/petstore/java/okhttp-gson-parcelableModel/src/main/java/org/openapitools/client/ApiClient.java

```diff
@@ -348,6 +345,16 @@ public Authentication getAuthentication(String authName) {
         return authentications.get(authName);
     }
 
+    /**
+    * Set default authentication.
+    *
+    * @param auth The authentication
+    */
+    public void setDefaultAuthentication(Authentication auth) {
+      authentications.put("default", auth);
```

Comment:
As discussed, the `default` can be already defined in the spec as other type of authentication such as HTTP basic, bearer, etc. Personally I don't prefer this logic or use case goes into the default template.

Have you considered using customized templates with the `-t` option instead?

Reply:
we'd prefer upstream the changes instead of using customized templates. what's your recommendation here? implement an Authentication that can refresh token instead of the default here?